handler/src/main/java/io/netty/handler/timeout/WriteTimeoutHandler.java

```diff
@@ -70,6 +70,8 @@
 
     private final long timeoutNanos;
 
+    private Set<ScheduledFuture<?>> timeouts;
```

Comment:
> no it's not... but it is guaranteed it is called once it is written or the write failed. I just wonder if we can somehow remove the overhead of the Set in terms of memory + remove/add.

So if just storing a reference to the last `timeout`, it may override the previous one. Then the previous one won't be cleaned in `handlerRemoved`. Seems we need a collection to store active timeouts at least.